src/Symfony/Component/Translation/Translator.php

```diff
@@ -158,7 +162,7 @@ public function setLocale(string $locale)
      */
     public function getLocale()
     {
-        return $this->locale;
+        return $this->locale ?: \Locale::getDefault();
```

Comment:
see also https://github.com/symfony/symfony/pull/38230, we'll introduce the same issue now :)